### ISSUE TITLE
Support Swift 3.1.1

### ISSUE THREAD

**Issue by tunniclm** [IBM-Swift/generator-swiftserver#182]
Update our generated `Package.swift` and generated Swift code to work with Swift 3.1.1.

We will skip Swift 3.1 as it has a crash bug that affects Kitura-WebSocket (which is used by SwiftMetrics and others).

**Comment by tunniclm** [comment]
We need all of the dependencies that we support to be able to compile and run on Swift 3.1.1, therefore we will update to the following dependency versions:

* `Kitura 1.6.x -> 1.7.x`
* `HeliumLogger 1.6.x -> 1.7.x`
* `Kitura-CouchDB 1.6.x -> 1.7.x`
* `Kitura-Redis 1.6.x -> 1.7.x`
* `CloudConfiguration 1.x -> 2.x`
* `Configuration 0.2.x -> 1.x`

Also, we will update the docker image tags:
* Dockerfile-tools: `ibmcom/swift-ubuntu 3.0.2 -> 3.1.1`
* Dockerfile: `ibmcom/swift-ubuntu-runtime 3.0.2 -> 3.1.1`

The Object Storage SDK (https://github.com/ibm-bluemix-mobile-services/bluemix-objectstorage-serversdk-swift) currently has a dependency on Kitura-net `1.6.x` via https://github.com/ibm-bluemix-mobile-services/bluemix-simple-http-client-swift which will need to move to Kitura-net `1.7.x` before we can continue.

**Comment by rob-deans** [comment]
Note - Update the *Resource.swift to silence the warnings with the new syntax

**Comment by tunniclm** [comment]
The Object Storage SDK has been updated, and we can now make the following additional dependency change:

* `bluemix-objectstorage-serversdk-swift 0.7.x -> 0.8.x`

However, I discovered that I has somehow missed the AppID SDK also needs an update, since it is currently dependent on `Kitura-Credentials 1.6.x` (which in turn depends on `Kitura 1.6.x` and `Kitura-Session 1.6.x`. Not sure how I missed this one before, since I thought the tests were passing on appid previously.

**Comment by tunniclm** [comment]
Looks like the AppID SDK just got updated with looser version requirements which will allow us to continue.

**Comment by tunniclm** [comment]
The build was failing since CloudConfiguration 2.x (edit: actually Swift-cfenv 4.x) doesn't support Swift 3.0.2. Updated CI to test on Swift 3.1.1 and now the tests are passing. I'd like to re-test once https://github.com/IBM-Swift/generator-swiftserver/pull/184 is merged, then this will be ready to go.

edit: opened issue for back-compatibility: https://github.com/IBM-Swift/Swift-cfenv/issues/59

**Comment by tunniclm** [comment]
Whilst the Swift 3.1.1 support has been in `develop` since May 5th, I have delayed cutting a release as I'd like to avoid a semver-major change here. Through some manual testing I have identified a few issues that would break compatibility with existing generated CRUD projects and I am addressing these in #199.

I would also like to get rid of the compile warnings related to `debugDescription` / `String(describing:)` before cutting the new release.
